hashicorp/react-components
# Implement redesign of `<Hero />`

🎟️ [Asana Task](https://app.asana.com/0/1201874072258661/1203137429559227/f)
🔍 [Preview Link](https://react-components-git-nahero-redesign-hashicorp.vercel.app/components/hero)

---



## Description
This PR updates the `<Hero />` to match the updated [WPL spec](https://www.figma.com/file/VvpEQaWhKQExx9QTWRyayd/Patterns?node-id=2591%3A14651&t=fQdpqIOCikoalpLj-1). This is both a big departure from the current implementation and not backwards compatible (thus the `major` bump in the changeset). Given this and its [usage on dot com](https://hashicorp-mktg.sourcegraph.com/search?q=context%3Aglobal+%40hashicorp%2Freact-hero&patternType=standard&groupBy=repo), the current implementation will be copied into dot com (likely as `<LegacyHero />`). 

While this isn't ideal, the component is used on numerous pages in ways that are not compatible with the newest version. Updating those instances would require a fair amount of coordination with various stakeholders and design to gracefully transition every page (gathering updated content / assets, ensuring the new component meets requirements, etc.).

## Validation Steps
- Navigate to [the preview](https://react-components-git-nahero-redesign-hashicorp.vercel.app/components/hero)
- In dev tools, delete the `sidebar` element and make the `stage` element span the full width of the page.
- Verify that the component matches the [spec](https://www.figma.com/file/VvpEQaWhKQExx9QTWRyayd/Patterns?node-id=2591%3A14651&t=fQdpqIOCikoalpLj-1).

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.
